build/scripts/generate_latest_metas.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+#
+# Copyright (c) 2018-2019 Red Hat, Inc.
+# This program and the accompanying materials are made
+# available under the terms of the Eclipse Public License 2.0
+# which is available at https://www.eclipse.org/legal/epl-2.0/
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+# Utility for automatically generating latest meta.yamls for plugins.
+#
+
+set -e
+
+for plugin_dir in v3/plugins/*/*; do
+  echo "Updating plugin ${plugin_dir}"
+  # Make sure every plugin has a 'latest.version' file
+  if [ ! -d "${plugin_dir}"/latest ] || [ ! -f "${plugin_dir}"/latest/latest.version ]; then
+    echo "    Missing latest.version: ${plugin_dir}/latest/latest.version"
+    exit 1
+  fi
+  # Generate meta.yaml for latest version
+  version=$(cat "${plugin_dir}/latest/latest.version")
+  latest_meta="${plugin_dir}/${version}/meta.yaml"
+  if [ ! -f "${latest_meta}" ]; then
+    echo "    [ERROR]: version.latest specifies '$version' but no such meta.yaml is found"
+    echo "             expecting: '${plugin_dir}/${version}/meta.yaml'"
+  fi
+  echo "    Found latest meta ${plugin_dir}/${version}/meta.yaml"
+  yq -y '. + {version: "latest"}' "$latest_meta" > "${plugin_dir}/latest/meta.yaml"
```

Comment:
I would also use a yq cleanup on `${plugin_dir}/${version}/meta.yaml` so both files are consistently formatted, including indents and quotes.

Reply:
There are two issues with this that leave me inclined to not modify existing metas:
1. We can't just copy back over since we've changed the version, we'd have to change the version back. `yq` also doesn't support in-place editing (so we can't easily do `yq -i '.' $file` to get what we want).
2. Modifying other files in the repo like this would mean anyone submitting a PR would have to run the `generate_latest_metas.sh` script before opening a PR, and we don't have a way to validate that. If a user does not run the script locally outside of the build for their PR, then the next person who does run it will have modified files spuriously.

Reply:
Re: 1, you lost me... we're not changing the version, we're simply reformatting the yaml

Re: 2, yes. They should run it. If they forget, the next person will automatically fix their mistake. If neither person runs it, the yaml is still valid and we're no worse than today. 

But having the script do the copy from /latest/ back to /version/ would allow the OPTION for them to see that the two files differ and can then choose to commit the updated /version/meta.yaml so it matches with /latest/meta.yaml.

IMHO there's no downside here.